astropy/modeling/astro_sherpa.py

```diff
@@ -0,0 +1,470 @@
+import numpy as np
+from collections import OrderedDict
+from sherpa.fit import Fit
+from sherpa.data import Data1D, Data1DInt, Data2D, Data2DInt, DataSimulFit
+from sherpa.models import UserModel, Parameter, SimulFitModel
+from .fitting import Fitter
+from ..utils.exceptions import AstropyUserWarning
+from sherpa.stats import Chi2, Chi2ConstVar, Chi2DataVar, Chi2Gehrels, Chi2ModVar, Chi2XspecVar, LeastSq
+from sherpa.optmethods import GridSearch, LevMar, MonCar, NelderMead
+from sherpa.estmethods import Confidence, Covariance, Projection
+# from astropy.modeling
+
+__all__ = ('SherpaFitter',)
+
+
+class SherpaFitter(Fitter):
+
+    """
+    Sherpa Fitter for astropy models. Yay :)
+
+    Parameters
+        ----------
+        optimizer : string
+            the name of a sherpa optimizer.
+        statistic : string
+            the name of a sherpa statistic.
+        estmethod : string
+            the name of a sherpa estmethod.
+    """
+
+    def __init__(self, optimizer="levmar", statistic="leastsq", estmethod="covariance"):
+        try:
+            optimizer = optimizer.optmethod
+        except AttributeError:
+            optimizer = OptMethod(optimizer).optmethod
```

Comment:
Oh, I see the definitions below. I would just change the order and define those classes first, because it makes reading the file from top to bottom easier.